### PR TITLE
chore: bump to Bazel 7

### PR DESCRIPTION
**Pull request by gregmagolan**
Pre-factor from https://github.com/aspect-build/rules_swc/pull/253.

Strange plugin test failure in #253 so I'm reducing the change set to narrow down what is causing it.

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -6,10 +6,10 @@ module(
     compatibility_level = 1,
 )
 
-bazel_dep(name = "aspect_bazel_lib", version = "1.30.2")
+bazel_dep(name = "aspect_bazel_lib", version = "1.42.3")
 
 # Note: only used for provider symbols, we don't spawn nodejs actions
-bazel_dep(name = "aspect_rules_js", version = "1.29.2")
+bazel_dep(name = "aspect_rules_js", version = "1.41.2")
```
Comment (gregmagolan) [inline MODULE.bazel:12]: Somewhere between 1.29.2 and latest release support for Bazel 7 was fixed....

File: MODULE.bazel [hunk 2]
```diff
@@ -6,10 +6,10 @@ module(
     compatibility_level = 1,
 )
 
-bazel_dep(name = "aspect_bazel_lib", version = "1.30.2")
+bazel_dep(name = "aspect_bazel_lib", version = "1.42.3")
```
Comment (gregmagolan) [inline MODULE.bazel:9]: rules_js 1.41.2 has a min of bazel-lib 1.42.3

File: MODULE.bazel (hunk 1 above)
Comment (alexeagle) [reply MODULE.bazel:12]: FYI https://pro.aspect.build/guides/bzlmod.pro#bazel-7-minimum-versions tells you exactly which version is minimum

Comment (gregmagolan) [reply MODULE.bazel:12]: Ahh. That makes it easier 👍

File: MODULE.bazel [hunk 3]
```diff
@@ -9,7 +9,7 @@ module(
 bazel_dep(name = "aspect_bazel_lib", version = "1.30.2")
 
 # Note: only used for provider symbols, we don't spawn nodejs actions
-bazel_dep(name = "aspect_rules_js", version = "1.29.2")
+bazel_dep(name = "aspect_rules_js", version = "1.31.0")
```
Comment (gregmagolan) [inline MODULE.bazel:12]: Minimum rules_js version for Bazel 7 support https://pro.aspect.build/guides/bzlmod.pro/#bazel-7-minimum-versions

File: MODULE.bazel [hunk 4]
```diff
@@ -6,12 +6,12 @@ module(
     compatibility_level = 1,
 )
 
-bazel_dep(name = "aspect_bazel_lib", version = "1.30.2")
+bazel_dep(name = "aspect_bazel_lib", version = "1.31.2")
```
Comment (gregmagolan) [inline MODULE.bazel:9]: `aspect_bazel_lib` & `platform` transitive bumps from the rules_js bump:
```
WARNING: For repository 'aspect_bazel_lib', the root module requires module version aspect_bazel_lib@1.30.2, but got aspect_bazel_lib@1.31.2 in the resolved dependency graph.
WARNING: For repository 'platforms', the root module requires module version platforms@0.0.4, but got platforms@0.0.7 in the resolved dependency graph.
```

File: swc/dependencies.bzl [hunk 5]
```diff
@@ -21,14 +21,14 @@ def rules_swc_dependencies():
 
     http_archive(
         name = "aspect_bazel_lib",
-        sha256 = "97fa63d95cc9af006c4c7b2123ddd2a91fb8d273012f17648e6423bae2c69470",
-        strip_prefix = "bazel-lib-1.30.2",
-        url = "https://github.com/aspect-build/bazel-lib/releases/download/v1.30.2/bazel-lib-v1.30.2.tar.gz",
+        sha256 = "0da75299c5a52737b2ac39458398b3f256e41a1a6748e5457ceb3a6225269485",
+        strip_prefix = "bazel-lib-1.31.2",
```
Comment (gregmagolan) [inline swc/dependencies.bzl:25]: Transtiive bump to match MODULE.bazel:
```
WARNING: For repository 'aspect_bazel_lib', the root module requires module version aspect_bazel_lib@1.30.2, but got aspect_bazel_lib@1.31.2 in the resolved dependency graph.
```

**Review by alexeagle — APPROVED** [review]

**Pull request merged**